src/account/account.entity.ts

```diff
@@ -1,161 +1,91 @@
 import { randomUUID } from 'node:crypto';
-import { BaseEntity } from '../core/base.entity';
 import { type CreatePostType, PostType } from '../post/post.entity';
-import type { Site } from '../site/site.service';
 
-export type PersistedAccount = Account & { id: number };
+export interface Account {
+    readonly id: number;
+    readonly uuid: string;
+    readonly username: string;
+    readonly name: string | null;
+    readonly bio: string | null;
+    readonly url: URL;
+    readonly avatarUrl: URL | null;
+    readonly bannerImageUrl: URL | null;
+    readonly apId: URL;
+    readonly apFollowers: URL | null;
+    readonly isInternal: boolean;
+    /**
+     * Returns a new Account instance which needs to be saved.
+     */
+    updateProfile(params: ProfileUpdateParams): Account;
+    /**
+     * @deprecated
```

Comment:
This code is weird and shouldn't really live here, or at least not be exposed like this - why does an Account have a `getApIdForPost` method lol